app/controllers/registrations_controller.rb

```diff
@@ -0,0 +1,43 @@
+class RegistrationsController < ApplicationController
+
+  def new
+    step_name = params[:step_name]
+    @registration = StepFactory.create(step_name)
+  rescue StepFactory::NameNotFoundError => e
+    # needs to redirect to root or last valid step
+    redirect_to :root, alert: "Error: #{e.message}"
+  end
+
+  def create
+    step_name = params[:step_name]
+    @registration = StepFactory.create(step_name)
+    @registration.assign_attributes(registration_params)
+  
+    if @registration.valid?
+      update_session_registration_hash
+      redirect_to new_registration_path(step_name: @registration.next_step)
+    else
+     render :new
+    end
+  end
+
+  private
+
+  def update_session_registration_hash
+    session[:registration] ||= {}
+    session[:registration].merge!(registration_params)
+  end
+
+  def entity_name
```

Comment:
Could be replaced with `@registration.step_name.to_sym`

Reply:
Good spot